src/software/ai/hl/stp/play/play.cpp

```diff
@@ -111,7 +111,7 @@ std::unique_ptr<TbotsProto::PrimitiveSet> Play::get(
     obstacle_list.Clear();
     path_visualization.Clear();
 
-    tactic_robot_id_assignment.clear();
+    std::map<std::shared_ptr<const Tactic>, RobotId> current_tactic_robot_id_assignment;
```

Comment:
shouldn't `RobotId` be the key for a map, or am I mixing up how syntax works...

Reply:
you would need to look into performance of swapping it or you already have - using tactics as keys implies that there is a constant set of tactics that doesn't change throughout an entire match, and the `RobotId` values are changing based on the situation.

Reply:
Using the tactic as the key is how the original authors of `Play` designed the tactic assignment map. Its not a great choice but I guess it works. Performance doesn't matter here since the number of tactics is small. 

I don't think using tactics as keys implies that there is a constant set of tactics. This function always gets a pointer to the same tactic instance while the tactic is executing (we only create new tactics when we switch plays or a tactic finishes). So by using the pointer as the key for the tactic instance, the tactic will always have the same key in the map while it's still running. Once the tactic finishes, it will be removed from the map (since we clear the map every tick and we've lost the pointer to the finished tactic). 

i.e. The keys in the map represent the tactic instances currently running.